python/ray/air/_internal/torch_utils.py

```diff
@@ -374,11 +374,24 @@ def arrow_batch_to_tensors(
         )
 
 
+def tensor_pin_memory_if_needed(tensor: torch.Tensor, pin_memory: bool) -> torch.Tensor:
+    if pin_memory and tensor.device.type == "cpu" and not tensor.is_pinned():
```

Comment:
I remember checking `tensor.is_pinned()` for each tensor is expensive. 
not entirely sure, we can do a microbenchmark to confirm
but if so, we should remove it.